src/parser/mod.rs

```diff
@@ -10408,13 +10408,86 @@ impl<'a> Parser<'a> {
             None
         };
 
+        let with_fill = if dialect_of!(self is ClickHouseDialect | GenericDialect)
+            && self.parse_keywords(&[Keyword::WITH, Keyword::FILL])
+        {
+            Some(self.parse_with_fill()?)
+        } else {
+            None
+        };
+
+        let interpolate = if dialect_of!(self is ClickHouseDialect | GenericDialect)
+            && self.parse_keyword(Keyword::INTERPOLATE)
+        {
+            if self.consume_token(&Token::LParen) {
+                if self.peek_token().token == Token::RParen {
```

Comment:
Oh actually to simplify the nested if/else, the interpolation parsing can use, `parse_comma_separated0` function instead, that one allows an empy list of arguments, seems like that would let us avoid the extra nested parens checks here?